src/python/pants/jvm/target_types.py

```diff
@@ -426,6 +436,127 @@ def validate(self) -> None:
+        cls, raw_value: dict[str, Iterable[str]], address: Address
+    ) -> FrozenDict[tuple[str, str], tuple[str, ...]]:
+        value_or_default = super().compute_value(raw_value, address)
+        assert value_or_default is not None
+        return FrozenDict(
+            {
+                cls._parse_coord(coord): tuple(packages)
+                for coord, packages in value_or_default.items()
+            }
+        )
+
+    @classmethod
+    def _parse_coord(cls, coord: str) -> tuple[str, str]:
+        group, artifact = coord.split(":")
+        return group, artifact
+
+
+class JvmArtifactsTargetGenerator(TargetGenerator):
+    alias = "jvm_artifacts"
+    core_fields = (
+        PomXmlSourceField,
+        JvmArtifactsPackageMappingField,
+        *COMMON_TARGET_FIELDS,
+    )
+    generated_target_cls = JvmArtifactTarget
+    copied_fields = COMMON_TARGET_FIELDS
+    moved_fields = (JvmArtifactResolveField,)
+    help = help_text(
+        """
+        Generate a `jvm_artifact` target for each dependency in pom.xml file.
+        """
+    )
+
+
+class GenerateFromPomXmlRequest(GenerateTargetsRequest):
+    generate_from = JvmArtifactsTargetGenerator
+
+
+@rule(
+    desc=("Generate `jvm_artifact` targets from pom.xml"),
+    level=LogLevel.DEBUG,
+)
+async def generate_from_pom_xml(
+    request: GenerateFromPomXmlRequest,
+    union_membership: UnionMembership,
+) -> GeneratedTargets:
+    generator = request.generator
+    pom_xml = await Get(
+        SourceFiles,
+        SourceFilesRequest([generator[PomXmlSourceField]]),
+    )
+    files = await Get(DigestContents, Digest, pom_xml.snapshot.digest)
+    mapping = request.generator[JvmArtifactsPackageMappingField].value
+    coordinates = parse_pom_xml(files[0].content)
+    targets = (
+        JvmArtifactTarget(
+            unhydrated_values={
+                "group": coord.group,
+                "artifact": coord.artifact,
+                "version": coord.version,
+                "packages": mapping.get((coord.group, coord.artifact)),
+                **request.template,
+            },
+            address=request.template_address.create_generated(coord.artifact),
+        )
+        for coord in coordinates
+    )
+    return GeneratedTargets(request.generator, targets)
+
+
+def parse_pom_xml(content: bytes) -> Iterator[Coordinate]:
+    root = ET.fromstring(content.decode("utf-8"))
+    match = re.match(r"^(\{.*\})project$", root.tag)
+    assert match, root.tag
+    namespace = match.group(1)
+    for dependency in root.iter(f"{namespace}dependency"):
+        yield Coordinate(
+            group=get_child_text(dependency, f"{namespace}groupId"),
+            artifact=get_child_text(dependency, f"{namespace}artifactId"),
+            version=get_child_text(dependency, f"{namespace}version"),
+        )
+
+
+def get_child_text(parent: ET.Element, child: str) -> str:
+    tag = parent.find(child)
+    if tag is None:
+        raise ValueError(f"missing tag: {child}")
```

Comment:
```suggestion
        raise ValueError(f"missing element: {child}")
```

The term is "element", not "tag" (tags mark the start and end of an element).  Yes, I am old enough to remember when XML was the new hot thing...